### ISSUE TITLE
Override scm vendor fetch (git, hg ect).

### ISSUE THREAD

**Issue by lateefj** [constabulary/gb#434]
I have a custom url that does not have git in the url that is hosting private repositories. I am working on a PR that would add this feature. As I am struggling to build / test I wanted to make sure something like this would be accepted before I put more time into it. Currently I was thinking something like this:
```
gb vendor fetch -scm git mysource.lhj.me/private/stuff
````

**Comment by davecheney** [comment]
Yeah, that sounds fine, but it will need to be a complete URL, Including the scheme.
